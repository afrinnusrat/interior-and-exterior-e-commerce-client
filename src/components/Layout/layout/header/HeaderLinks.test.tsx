import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { _HeaderLinks, HeaderLinksProps } from './HeaderLinks.component';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(cleanup);

const withRouter = (Component: any) => (props: any) => (
  <Router>
    <Component {...props} />
  </Router>
);

const HeaderLinksWithRouter = withRouter(_HeaderLinks);

describe('<HeaderLinks />', () => {
  const mockCheckSession = jest.fn();
  const mockFetchProductsByType = jest.fn();

  let props: HeaderLinksProps = {
    isAuth: false,
    cartItemsCount: 2,
    checkSession: mockCheckSession,
    fetchProductsByType: mockFetchProductsByType
  };

  it('renders correctly', () => {
    const { asFragment } = render(HeaderLinksWithRouter(props));
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls `fetchProductsByType` action with `all` arg', () => {
    const { getByTestId } = render(HeaderLinksWithRouter(props));
    fireEvent.click(getByTestId('PageLink'));
    expect(mockFetchProductsByType).toBeCalledWith('all');
  });

  it('calls `fetchProductsByType` action with `clocks` arg', () => {
    const { getByTestId } = render(HeaderLinksWithRouter(props));
    fireEvent.click(getByTestId('Clocks'));
    expect(mockFetchProductsByType).toBeCalledWith('clocks');
  });

  it('renders `SignOutBtn` when user is authenticated', () => {
    const { getByTestId } = render(
      HeaderLinksWithRouter({ ...props, isAuth: true })
    );
    expect(getByTestId('SignOutBtn')).toBeInTheDocument();
  });

  it('calls `chekcSession` action', () => {
    const { getByTestId } = render(
      HeaderLinksWithRouter({ ...props, isAuth: true })
    );
    fireEvent.click(getByTestId('SignOutBtn'));
    expect(mockCheckSession).toHaveBeenCalled();
  });
});
