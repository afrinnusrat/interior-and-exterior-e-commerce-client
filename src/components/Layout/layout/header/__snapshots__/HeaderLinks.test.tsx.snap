// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLinks /> renders correctly 1`] = `
<DocumentFragment>
  <ul
    class="sc-bdVaJa hDTYr"
  >
    <li
      class="sc-bxivhb icElMN"
    >
      <a
        class="sc-htpNat gyqNkk"
        data-testid="PageLink"
        href="/shop"
      >
        Products
        <i
          class="fas fa-caret-down"
        />
      </a>
      <div
        class="sc-bwzfXH gjnxxD"
      >
        <ul
          class="sc-ifAKCX kqqTha"
        >
          <li
            class="sc-EHOje jTwYBT"
          >
            <a
              class="sc-bZQynM imRHDl"
              data-testid="Chairs"
              href="/shop"
            >
              Chairs
            </a>
          </li>
          <li
            class="sc-EHOje jTwYBT"
          >
            <a
              class="sc-bZQynM imRHDl"
              data-testid="Lamps"
              href="/shop"
            >
              Lamps
            </a>
          </li>
          <li
            class="sc-EHOje jTwYBT"
          >
            <a
              class="sc-bZQynM imRHDl"
              data-testid="Clocks"
              href="/shop"
            >
              Clocks
            </a>
          </li>
          <li
            class="sc-EHOje jTwYBT"
          >
            <a
              class="sc-bZQynM imRHDl"
              data-testid="Vases"
              href="/shop"
            >
              Vases
            </a>
          </li>
          <li
            class="sc-EHOje jTwYBT"
          >
            <a
              class="sc-bZQynM imRHDl"
              data-testid="Benches"
              href="/shop"
            >
              Benches
            </a>
          </li>
          <li
            class="sc-EHOje jTwYBT"
          >
            <a
              class="sc-bZQynM imRHDl"
              data-testid="Flowerpots"
              href="/shop"
            >
              Flowerpots
            </a>
          </li>
          <li
            class="sc-EHOje jTwYBT"
          >
            <a
              class="sc-bZQynM imRHDl"
              data-testid="Plants"
              href="/shop"
            >
              Plants
            </a>
          </li>
        </ul>
      </div>
    </li>
    <li
      class="sc-bxivhb icElMN"
    >
      <a
        class="sc-htpNat gyqNkk"
        href="/account"
      >
        Account
      </a>
    </li>
    <li
      class="sc-bxivhb icElMN"
    >
      <a
        class="sc-htpNat gyqNkk"
        href="/cart"
      >
        Cart (2)
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
